packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -1023,34 +1034,53 @@ class _IconButtonDefaultForeground extends MaterialStateProperty<Color?> {
 
 @immutable
 class _IconButtonDefaultOverlay extends MaterialStateProperty<Color?> {
-  _IconButtonDefaultOverlay(this.foregroundColor, this.focusColor, this.hoverColor, this.highlightColor);
+  _IconButtonDefaultOverlay(
+    this.foregroundColor,
+    this.focusColor,
+    this.hoverColor,
+    this.highlightColor,
+    this.overlayColor,
+  );
 
   final Color? foregroundColor;
   final Color? focusColor;
   final Color? hoverColor;
   final Color? highlightColor;
+  final Color? overlayColor;
 
   @override
   Color? resolve(Set<MaterialState> states) {
     if (states.contains(MaterialState.selected)) {
       if (states.contains(MaterialState.pressed)) {
-        return highlightColor ?? foregroundColor?.withOpacity(0.1);
+        return highlightColor
```

Comment:
This is fine as is. I suppose you could have also written (here and elsewhere):
```dart
return highlightColor ?? (overlayColor ?? foregroundColor)?.withOpacity(0.1);
```

Reply:
Perfect!